src/Microsoft.Azure.WebJobs.Host/Hosting/ServiceCollectionDecorator.cs

```diff
@@ -0,0 +1,100 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.Azure.WebJobs.Host.Hosting
+{
+    internal class ServiceCollectionDecorator : IServiceCollection, ITrackedServiceCollection
+    {
+        public IServiceCollection ServiceCollection { get; set; }
+
+        public IEnumerable<ServiceDescriptor> TrackedCollectionChanges // tracks any service collection modification.
+        {
+            get
+            {
+                return _trackedChanges as IEnumerable<ServiceDescriptor>;
+            }
+            private set { }
+        }
```

Comment:
For this you should be able to just do (it just makes it shorter).

` public IEnumerable<ServiceDescriptor> TrackedCollectionChanges => _trackedChanges;`